tools/mlflow_query.py

```
@@ -0,0 +1,564 @@
+        filter_string="tags.run_type = 'parent'",
+        order_by=["start_time DESC"],
+        max_results=1000,
+    )
+    for run in runs:
+        if run.info.run_name == name_or_id:
+            return run
+
+    # Fallback: treat argument as a raw run ID
+    try:
+        return client.get_run(name_or_id)
+    except Exception as e:
```

Ruff (0.15.10)

[warning] Do not catch blind exception: `Exception`

(BLE001)